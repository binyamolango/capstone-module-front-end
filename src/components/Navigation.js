import { NavLink, useNavigate } from 'react-router-dom';
import { useState, useEffect } from 'react';
import MenuIcon from '@mui/icons-material/Menu';
import CloseIcon from '@mui/icons-material/Close';
import WavingHandIcon from '@mui/icons-material/WavingHand';
import logo from '../images/logo.png';
import fb from '../images/fb.svg';
import google from '../images/google.svg';
import x from '../images/x.svg';
import github from '../images/github.svg';
import linkedin from '../images/linkedin.svg';

const Navigation = () => {
  const [showMenu, setShowMenu] = useState(false);
  const navigate = useNavigate();
  const [name, setName] = useState('');

  useEffect(() => {
    const user = JSON.parse(localStorage.getItem('user'));
    setName(user);
  }, []);

  const toggleMenu = () => {
    setShowMenu(!showMenu);
  };

  const closeMenu = () => {
    setShowMenu(false);
  };
  const handleLogout = () => {
    localStorage.removeItem('user');
    navigate('/');
    window.location.reload();
  };

  return (
    <div className="nav__section">
      <button
        type="button"
        className="md:hidden absolute top-4 left-4 mr-4 p-2 text-gray-500 hover:text-gray-700 focus:outline-none focus:ring focus:ring-gray-200"
        onClick={toggleMenu}
      >
        {showMenu ? <CloseIcon /> : <MenuIcon />}
      </button>

    </div>
  );
};

export default Navigation;
