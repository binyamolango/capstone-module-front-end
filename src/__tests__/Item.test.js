import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Navigation from '../components/Navigation';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => jest.fn(),
}));

const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUseNavigate,
}));

describe('Tests Navigation component', () => {
  it('should render page', () => {
    const NavigationComponent = renderer.create(
      <BrowserRouter>
        <Navigation />
      </BrowserRouter>,
    );
    expect(NavigationComponent).toMatchSnapshot();
  });

  it('should display navigation buttons "Doctors"', () => {
    render(
      <BrowserRouter>
        <Navigation />
      </BrowserRouter>,
    );
    const button1 = screen.getByText('Delete Doctor');
    expect(button1).toBeInTheDocument();
  });

  it('should display navigation buttons "Add Doctor"', () => {
    render(
      <BrowserRouter>
        <Navigation />
      </BrowserRouter>,
    );
    const button1 = screen.getByText('Add Doctor');
    expect(button1).toBeInTheDocument();
  });

  it('should display navigation buttons "Doctors"', () => {
    render(
      <BrowserRouter>
        <Navigation />
      </BrowserRouter>,
    );
    const button1 = screen.getByText('Doctors');
    expect(button1).toBeInTheDocument();
  });

  it('should display navigation buttons "My Appointments"', () => {
    render(
      <BrowserRouter>
        <Navigation />
      </BrowserRouter>,
    );
    const button1 = screen.getByText('My Appointments');
    expect(button1).toBeInTheDocument();
  });

  it('should display navigation buttons "Book Appointment"', () => {
    render(
      <BrowserRouter>
        <Navigation />
      </BrowserRouter>,
    );
    const button1 = screen.getByText('Book Appointment');
    expect(button1).toBeInTheDocument();
  });
});
