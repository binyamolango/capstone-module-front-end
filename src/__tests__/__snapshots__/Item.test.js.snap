// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Item component should render page 1`] = `
<div
  className="flex items-center justify-center h-screen"
>
  <span
    className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </span>
</div>
`;

exports[`Tests Navigation component should render page 1`] = `
<div
  className="nav__section"
>
  <button
    className="md:hidden absolute top-4 left-4 mr-4 p-2 text-gray-500 hover:text-gray-700 focus:outline-none focus:ring focus:ring-gray-200"
    onClick={[Function]}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="MenuIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
      />
    </svg>
  </button>
  <div
    className="hidden md:flex flex-col pt-4 pb-12 md:pb-1 justify-between items-center h-[100dvh] bg-white  w-full drop-shadow-md"
  >
    <div>
      <img
        alt=""
        className="logo-img"
        src="logo.png"
      />
      <button
        className="md:hidden absolute top-4 left-4 mr-4 p-2 text-gray-500 hover:text-gray-700 focus:outline-none focus:ring focus:ring-gray-200"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
          />
        </svg>
      </button>
    </div>
    <div
      className="-m-20 text-center w-full uppercase"
    >
      <h1
        className="font-medium w-full p-2"
      >
        Hello
         
        <span
          className="text-[#4e8de0]"
        >
           
        </span>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="WavingHandIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M23 17c0 3.31-2.69 6-6 6v-1.5c2.48 0 4.5-2.02 4.5-4.5zM1 7c0-3.31 2.69-6 6-6v1.5C4.52 2.5 2.5 4.52 2.5 7zm7.01-2.68-4.6 4.6c-3.22 3.22-3.22 8.45 0 11.67s8.45 3.22 11.67 0l7.07-7.07c.49-.49.49-1.28 0-1.77s-1.28-.49-1.77 0l-4.42 4.42-.71-.71 6.54-6.54c.49-.49.49-1.28 0-1.77s-1.28-.49-1.77 0l-5.83 5.83-.71-.71 6.89-6.89c.49-.49.49-1.28 0-1.77s-1.28-.49-1.77 0l-6.89 6.89-.69-.7 5.48-5.48c.49-.49.49-1.28 0-1.77s-1.28-.49-1.77 0l-7.62 7.62c1.22 1.57 1.11 3.84-.33 5.28l-.71-.71c1.17-1.17 1.17-3.07 0-4.24l-.35-.35 4.07-4.07c.49-.49.49-1.28 0-1.77-.5-.48-1.29-.48-1.78.01"
          />
        </svg>
      </h1>
      <p
        className="font-medium text-sm w-full"
      >
        Welcome to
        <span
          className="text-[#4e8de0]"
        >
           our system
        </span>
      </p>
    </div>
    <nav
      className="w-full"
    >
      <ul
        className="flex flex-col"
      >
        <a
          className="active-link"
          href="/"
          onClick={[Function]}
        >
          <li
            className="flex p-4 w-full text-sm font-medium cursor-pointer border-b border-gray-200"
          >
            Doctors
          </li>
        </a>
        <a
          className="active-link"
          href="/bookappointment"
          onClick={[Function]}
        >
          <li
            className="flex p-4 w-full text-sm font-medium cursor-pointer border-b border-gray-200"
          >
            Book Appointment
          </li>
        </a>
        <a
          className="active-link"
          href="/myappointments"
          onClick={[Function]}
        >
          <li
            className="flex p-4 w-full text-sm font-medium cursor-pointer border-b border-gray-200"
          >
            My Appointments
          </li>
        </a>
        <a
          className="active-link"
          href="/adddoctor"
          onClick={[Function]}
        >
          <li
            className="flex p-4 w-full text-sm font-medium cursor-pointer border-b border-gray-200"
          >
            Add Doctor
          </li>
        </a>
        <a
          className="active-link"
          href="/deletedoctor"
          onClick={[Function]}
        >
          <li
            className="flex p-4 w-full text-sm font-medium cursor-pointer border-b border-gray-200"
          >
            Delete Doctor
          </li>
        </a>
      </ul>
      <div
        className="text-center m-3 pt-2"
      >
        <button
          className="pl-6 pr-6 pt-1 pb-1 self-end text-white bg-blue-400 rounded-r-[100px] rounded-l-[100px]"
          onClick={[Function]}
          type="button"
        >
          Logout
        </button>
      </div>
    </nav>
    <footer
      className="footer flex flex-col gap-4 w-full pt-4 pb-2 pl-4 pr-4"
    >
      <ul
        className="footer-center justify-start items-end gap-3.5 inline-flex"
      >
        <li
          className="w-6 h-6 relative"
        >
          <img
            alt=""
            src="fb.svg"
          />
        </li>
        <li
          className="w-6 h-6 relative"
        >
          <img
            alt=""
            src="x.svg"
          />
        </li>
        <li
          className="w-6 h-6 relative"
        >
          <img
            alt=""
            src="google.svg"
          />
        </li>
        <li
          className="w-6 h-6 relative"
        >
          <img
            alt=""
            src="github.svg"
          />
        </li>
        <li
          className="w-6 h-6 relative"
        >
          <img
            alt=""
            src="linkedin.svg"
          />
        </li>
      </ul>
      <p
        className="text-gray-300 text-xs font-medium font-['Inter'] leading-[18px] tracking-[2.88px]"
      >
        @ 2024 Edenic Health
      </p>
    </footer>
  </div>
</div>
`;
